src/jshint.js

```diff
@@ -2508,22 +2508,26 @@ var JSHINT = (function() {
       return doFunction({ type: "arrow", parsedOpening: true });
     }
 
-    var exprs = [];
+    if (state.tokens.next.id === ")") {
```

Comment:
So there are a few things here:

first off: lets put this check behind an enforcing flag, so that this is not actually a breaking change. We could allow `()` by default, but error if an option is enabled. This could later be turned into a leniency option in the next version.

second: `() => { /* do stuff */ }` is a valid and useful pattern, so we'll also want to make sure it's actually an empty expression, and not [ArrowFormalParameters](https://tc39.github.io/ecma262/#prod-CoverParenthesizedExpressionAndArrowParameterList).

Reply:
ah, after reading the code a bit more, that second point seems to be irrelevant. so, lgtm but I think we still want a flag here to avoid the breaking change.